sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -1385,8 +1389,49 @@ case class DropIndex(
 case class TableSpec(
     properties: Map[String, String],
     provider: Option[String],
-    options: Map[String, String],
+    // Note that the map value can be null if the input option only provided a key.
+    optionsList: Map[String, Expression],
     location: Option[String],
     comment: Option[String],
     serde: Option[SerdeInfo],
-    external: Boolean)
+    external: Boolean) extends TreePatternBits {
+  /** This is a convenience method to obtain the options list as a map of strings to strings. */
+  lazy val options: Map[String, String] = {
```

Comment:
This looks hacky. Ideally, a SQL query should work even when the optimizer rule `ConstantFolding`.

Sorry for getting back and forth: shall we have a `UnresolvedTableSpec` which contains `optionsList: Map[String, Expression]`. And another rule to resolve `UnresolvedTableSpec` as `TableSpec` if the expression in the map is all resolved? We can try using expression.eval() in the new rule.

Reply:
Sure, this is done.